mscore/plugin/api/cursor.cpp

```diff
@@ -127,13 +135,26 @@ void Cursor::rewind(RewindMode mode)
 
 bool Cursor::next()
       {
-      if (!_segment)
-            return false;
-      _segment = _segment->next1(_filter);
-      nextInTrack();
-      _score->inputState().setTrack(_track);
-      _score->inputState().setSegment(_segment);
-      return _segment != 0;
+      if (_elemListIdx >= 0)
```

Comment:
Concerning the selection-related changes, while they are definitely useful I believe that `next()` function is not the correct place to do this as this function is supposed to move to the next segment, and making this dependent on the rewind mode could complicate things unnecessarily. Maybe the most natural solution would be just to expose the `Selection` object from a score — for now only some properties in a read-only way, potentially that could include some possibilities to change a selection programmatically.
Some very brief discussion of exposing selection to plugins API can also be found in [this thread](https://musescore.org/en/node/282107) (mostly [this comment](https://musescore.org/en/node/282107#comment-889370)).

Also it would be preferable to implement this change as a separate pull request, this would make reviewing and testing the particular changes easier.

Reply:
>Also it would be preferable to implement this change as a separate pull request, this would make reviewing and testing the particular changes easier.

Agreed. For now we've been testing it as a package deal for the issue. I'll do that as soon as an approach is settled on.

I will note that in your linked comment it was suggested that:

"Still, I think I'd like the plugin API to be more functionality-oriented (e.g. by extending the Cursor class) "

Seems like I wasn't so far off base with my approach. :) It just pointed a different things.

-Dale

Reply:
I've tried to expose the Selection object and it seemed straight forward but I been odd behaviors between the Cursor approach and the Selection approach. 

When the Cursor approach is used I can manipulate Note.playEvents all I want and it works great!

When I use the Selection object to access the very same intern list of Elements it deliver's Notes but QML doesn't believe the Note even has a playEventList.

I've pushed up commits that contain both approaches so they can be debugged at the same time. I also pushed up a commit containing the QML test scripts and score I've been using for testing.

I can't figure out why the Note.playEvents are hidden when using Selection and not when using Cursor.

HELP NEEDED!

-Dale

Reply:
I believe the reason of this is that `wrapContainerProperty<Element>()` simply wraps the elements to `Element` type wrapper while it would be better to determine the correct type of a wrapper object based on the wrapped element's type. It shouldn't be difficult to correct, I'll post a patch for it tomorrow.

Reply:
Bless you sir!
-Dale

Reply:
Something like this seems to correct the issue: https://github.com/dmitrio95/MuseScore/commit/344b1f3202a8463182acb384bf59f470851e0215.

Reply:
That indeed fixed the problem. It looked to be a bit more subtle than I expected.

Thanks a bunch! I'll start the PR clean-up and final items now.

-Dale

Reply:
New coded submitted.